workflow/controller/workflowpod.go

```diff
@@ -375,7 +383,12 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 		pod.Spec.ActiveDeadlineSeconds = &newActiveDeadlineSeconds
 	}
 
-	created, err := woc.controller.kubeclientset.CoreV1().Pods(woc.wf.ObjectMeta.Namespace).Create(pod)
+	var created *apiv1.Pod
+	err = retryutil.OnError(createPodRetryBackoff, woc.shouldRetryCreate, func() error {
```

Comment:
I'm not sure the best solution here. We have a limited time (30s) seconds to reconcile a workflow. I wonder if it would be better rather that fail the workflow on transient error - to instead exit early and wait for the next reconciliation?

Reply:
That makes sense. I update the PR and included server timeout and also service unavailable in `IsTransientErr`.